core/core-spacemacs-buffer.el

```diff
@@ -828,9 +826,53 @@ list. Return entire list if `END' is omitted."
                   (when (spacemacs-buffer//insert-file-list
                          "Projects:"
                          (spacemacs//subseq (projectile-relevant-known-projects)
-                                     0 list-size))
+                                            0 list-size))
                     (spacemacs//insert--shortcut "p" "Projects:")
-                    (insert list-separator)))))) dotspacemacs-startup-lists))))
+                    (insert list-separator)))))) dotspacemacs-startup-lists)))
+
+(defun spacemacs-buffer//get-buffer-width ()
+  (save-excursion
+    (goto-char 0)
+    (let ((current-max 0))
+      (while (not (eobp))
+        (let ((line-length (- (line-end-position) (line-beginning-position))))
+          (if (< current-max line-length)
+              (setq current-max line-length)))
+        (forward-line 1))
+      current-max)))
+
+(defvar spacemacs-buffer--buttons-position nil
+  "Offset in characters between the edge of the screen and the beginning of the
+ home buffer buttons. Do not try to set this variable.")
+
+(defvar spacemacs-buffer-buttons-recent-lists-offset 25
```

Comment:
Yes, it is independent of the width of the startup lists.  
The reason is that just centering based on the width of the startup lists would feel weird, so I did not just center it.  

The positioning algorithm is as follows:
- if there is enough space, the startup lists will have a fixed relative position to the actually centered elements (here the home buffer buttons)
- if a part of the startup lists would be off the screen just center them regularly, based on the width of the startup lists and the width of the window.

This way, the position of the beginning of the lines is consistent which makes reading easier, and the width of the buffer is always used at its best even when some entries are very long.